src/components/tracked-controls.js

```diff
@@ -0,0 +1,149 @@
+var registerComponent = require('../core/component').registerComponent;
+var THREE = require('../lib/three');
+
+module.exports.Component = registerComponent('tracked-controls', {
+  schema: {
+    controller: { default: 0 }
+  },
+
+  init: function () {
+    this.buttonStates = {};
+    this.axisState = [];
+  },
+
+  update: function () {
+    var controllers = this.system.controllers;
+    // handId: 0 - right, 1 - left
+    this.controller = controllers[this.data.controller];
+  },
+
+  tick: function (time, delta) {
+    var mesh = this.el.getObject3D('mesh');
+    if (!mesh) { return; }
+    if (mesh.update) { mesh.update(delta / 1000); }
+    this.updatePose();
+    this.updateButtons();
+  },
+
+  updatePose: (function () {
+    var controllerEuler = new THREE.Euler();
+    var controllerPosition = new THREE.Vector3();
+    var controllerQuaternion = new THREE.Quaternion();
+    var dolly = new THREE.Object3D();
+    var standingMatrix = new THREE.Matrix4();
+    controllerEuler.order = 'YXZ';
+    return function () {
+      var controller;
+      var pose;
+      var orientation;
+      var position;
+      var el = this.el;
+      var vrDisplay = this.system.vrDisplay;
+      this.update();
+      controller = this.controller;
+      if (!controller) { return; }
+      pose = controller.pose;
+      orientation = pose.orientation || [0, 0, 0, 1];
+      position = pose.position || [0, 0, 0];
```

Comment:
I don't see the spec related to the Gamepad extension but I would assume that would follow the same convention than the VRDisplay interface:

https://w3c.github.io/webvr/#dom-vrdisplay-getpose